www/pfSense-pkg-vHosts/files/usr/local/pkg/vhosts.inc

```diff
@@ -0,0 +1,367 @@
+
+        @custom_config
+
+        location / { index  index.html index.htm index.php; }
+
+        location ~ \.php$
+            {
+            try_files            \$uri =404; # This line closes a potential security hole
+                                             # ensuring users can't execute uploaded files
+                                             # see: http://forum.nginx.org/read.php?2,88845,page=3
+            fastcgi_pass         unix:/var/run/php-fpm.socket;
+            fastcgi_index        index.php;
+            fastcgi_param        SCRIPT_FILENAME  \$document_root\$fastcgi_script_name;
+            fastcgi_param        HTTP_PROXY  ""; # Fix httpoxy - https://httpoxy.org/#fix-now
+            fastcgi_read_timeout 180;
+            include              /usr/local/etc/nginx/fastcgi_params;
+            }
+
+        location = /favicon.ico  # Do not log requests for favicon.ico.
+            {
+            return               204;
+            access_log           off;
+            log_not_found        off;
+            }
+        }
+CONF_SERVER;
+
+/*-----------------------------------------*/
+/* nginx server SSL config block template. */
+/*                                         */
+/* @crt_file_path = Certificate file path. */
+/* @key_file_path = Private key file path. */
+/*-----------------------------------------*/
+$vhosts_g['conf_server_ssl'] = <<<CONF_SERVER_SSL
+
+        ssl_certificate                         '@crt_file_path';
+        ssl_certificate_key                     '@key_file_path';
+        ssl_session_timeout                     10m;
+        keepalive_timeout                       70;
+        ssl_session_cache                       shared:SSL:10m;
+        ssl_protocols                           TLSv1 TLSv1.1 TLSv1.2;
+        ssl_ciphers                             "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH";
+        ssl_prefer_server_ciphers               on;
+        add_header Strict-Transport-Security    "max-age=31536000";
+        add_header X-Content-Type-Options       nosniff;
+        ssl_session_tickets                     off;
+        ssl_dhparam                             /etc/dh-parameters.4096;
+CONF_SERVER_SSL;
+
+/*******************************************************************************
+* vhosts_build_service_config */
+/**
+* Rebuilds nginx configuration for vhosts. Called for package synchronization
+* and when any vhost is modified.
+*
+* Config file setting: <custom_php_resync_config_command>
+*******************************************************************************/
+function vhosts_build_service_config() {
+	global $config, $vhosts_g;
+
+	$conf_file_path   = $vhosts_g['conf_file_path'];
+	$conf_file_prefix = $vhosts_g['conf_file_prefix'];
+
+	/*-------------------------------------*/
+	/* Remove all existing configurations. */
+	/*-------------------------------------*/
+	conf_mount_rw();
+	exec("/bin/rm -f $conf_file_prefix*");
```

Comment:
I know it's one you set manually as a global, but please make sure `$conf_file_prefix` contains a sane value before passing it to things like `rm -f` with a glob. Also, when using variables inside strings, surround the variable with braces to make the separation clear, e.g. `/bin/rm -f {$conf_file_prefix}*` -- Otherwise, especially in cases like this, it is easy to overlook the `*`.

In this specific instance, using our [unlink_if_exists()](https://github.com/pfsense/pfsense/blob/master/src/etc/inc/util.inc#L1712) function would be best, it can accept a glob as a parameter and doesn't require a shell exec.